### ISSUE TITLE
Debian Security Advisory package builds are not incorporated ASAP

### ISSUE THREAD

**Issue by jonnyowenpowell** [GoogleContainerTools/distroless#1231]
**Describe the bug**
As occurred in #1229, the new version of an updated package is required by the CI workflow to be present for all architectures.

Debian Security Advisories are sometimes published without a build of the package for some of the less common architectures, to secure a large percentage of their users sooner. See details [here](https://www.debian.org/security/faq#archismissing).

By requiring builds be present for all architectures, distroless currently breaks the ASAP delivery of the fix.

**To Reproduce**

An example of this can be seen in #1229, when [DSA-5343-1](https://www.debian.org/security/2023/dsa-5343) was released without a build for armhf. This delayed the release of the fix for other architectures by ~24 hours.

**Expected behavior**

Package builds released with a DSA are incorporated into distroless for each architecture independently.

**Console Output**
If applicable, add information from your container run

**Additional context**
There is a [DSA mailing list](https://lists.debian.org/debian-security-announce) which could be useful for this work.

**Comment by loosebazooka** [comment]
Yeah this is a known issue that we occasionally run into. The "package manager" needs a rework.

**Comment by loosebazooka** [comment]
I think this is solved by the new rules_distroless handling
